--- /import/lab/users/sanchez/Desktop/ADSProject_Master/04_forwarding_unit/src/main/scala/IDbarrier.scala	2026-06-24 15:48:23.000000000 +0200
+++ course_material/task4_level2/solution/src/main/scala/IDbarrier.scala	2026-07-01 18:17:00.454385046 +0200
@@ -42,6 +42,8 @@
 class IDBarrier extends Module {
   val io = IO(new Bundle {
     val inUOP          = Input(uopc())
+    val inInstr        = Input(UInt(32.W))
+    val inPC           = Input(UInt(32.W))
     val inRS1          = Input(UInt(5.W))          
     val inRS2          = Input(UInt(5.W))
     val inRD           = Input(UInt(5.W))
@@ -50,6 +52,8 @@
     val inWrEn         = Input(Bool())
     val inXcptInvalid  = Input(Bool())
     val outUOP         = Output(uopc())
+    val outInstr       = Output(UInt(32.W))
+    val outPC          = Output(UInt(32.W))
     val outRS1         = Output(UInt(5.W))          
     val outRS2         = Output(UInt(5.W))
     val outRD          = Output(UInt(5.W))
@@ -60,6 +64,8 @@
   })
 
   val uop         = Reg(uopc())
+  val instr       = RegInit(0.U(32.W))
+  val pc          = RegInit(0.U(32.W))
   val rs1         = RegInit(0.U(5.W))
   val rs2         = RegInit(0.U(5.W))
   val rd          = RegInit(0.U(5.W))
@@ -71,6 +77,12 @@
   io.outUOP := uop
   uop := io.inUOP
 
+  io.outInstr := instr
+  instr := io.inInstr
+
+  io.outPC := pc
+  pc := io.inPC
+
   io.outRS1 := rs1
   rs1 := io.inRS1
 
