--- /import/lab/users/sanchez/Desktop/ADSProject_Master/04_forwarding_unit/src/main/scala/RegisterFile.scala	2026-06-24 15:48:23.000000000 +0200
+++ course_material/task4_level2/solution/src/main/scala/RegisterFile.scala	2026-07-01 18:17:34.978688838 +0200
@@ -56,6 +56,8 @@
     val req_2  = new regFileReadReq
     val resp_2 = new regFileReadResp
     val req_3  = new regFileWriteReq
+    val gpRegVal = Output(UInt(32.W))
+    val debug_regs = Output(Vec(32, UInt(32.W)))
 })
 
   val regFile = Mem(32, UInt(32.W))
@@ -69,5 +71,9 @@
 
   io.resp_1.data := Mux(io.req_1.addr === io.req_3.addr, io.req_3.data, Mux(io.req_1.addr === 0.U, 0.U, regFile(io.req_1.addr)))
   io.resp_2.data := Mux(io.req_2.addr === io.req_3.addr, io.req_3.data, Mux(io.req_2.addr === 0.U, 0.U, regFile(io.req_2.addr)))
+  io.gpRegVal := regFile(3)
+  for (i <- 0 until 32) {
+    io.debug_regs(i) := Mux(i.U === 0.U, 0.U, regFile(i))
+  }
 
 }
