--- /import/lab/users/sanchez/Desktop/ADSProject_Master/04_forwarding_unit/src/main/scala/WBbarrier.scala	2026-06-24 15:48:23.000000000 +0200
+++ course_material/task4_level2/solution/src/main/scala/WBbarrier.scala	2026-07-01 18:17:25.946605922 +0200
@@ -36,16 +36,28 @@
   val io = IO(new Bundle {
     val inCheckRes    = Input(UInt(32.W))
     val outCheckRes   = Output(UInt(32.W))
+    val inInstr        = Input(UInt(32.W))
+    val outInstr       = Output(UInt(32.W))
+    val inPC           = Input(UInt(32.W))
+    val outPC          = Output(UInt(32.W))
     val inXcptInvalid = Input(Bool())
     val outXcptInvalid  = Output(Bool())
   })
 
   val check_res   = RegInit(0.U(32.W))
+  val instr       = RegInit(0.U(32.W))
+  val pc          = RegInit(0.U(32.W))
   val XcptInvalid = RegInit(false.B)
 
   io.outCheckRes := check_res
   check_res      := io.inCheckRes
 
+  io.outInstr := instr
+  instr       := io.inInstr
+
+  io.outPC := pc
+  pc       := io.inPC
+
   io.outXcptInvalid := XcptInvalid
   XcptInvalid := io.inXcptInvalid
 }
